lib/watch/fork.js

```diff
@@ -0,0 +1,35 @@
+const chalk = require('chalk')
+const { stop, runFastify } = require('../../start')
+
+const GRACEFUL_SHUT = 'GRACEFUL SHUTDOWN'
+const READY = 'ready'
+
+const fastify = runFastify(process.argv.splice(2))
+const type = process.env.childEvent
+
+process.send({ type: type, err: null })
+
+fastify.ready(err => {
+  process.send({ type: READY, err: err })
+  if (err) { stop(err) }
+})
+
+process.on('message', function (event) {
+  if (event === GRACEFUL_SHUT) {
+    const timeout = setTimeout(function () {
+      process.exit(1)
+    }, 5000)
+
+    fastify.close(() => {
+      timeout.unref()
```

Comment:
It fastify has provided the hook function of before .close() ? I think it's unref() should call fastify.close() before or fastify.close() callback。

Reply:
Calling `unref()` before `process.exit(0)` is useless (we will exit anyway). If for some reason an application is not calling the `onClose()` callbacks, this will hang for 5s. calling `unref()` would make it possible for that application to exit normally.

Reply:
thank you very much. I get it.